src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -6704,6 +6720,705 @@ namespace HeatBalanceSurfaceManager {
+        // surface in the building. It is a copy of CalcHeatBalanceInsideSurf,
+        // simplified for CTF surfaces only.
+
+        // REFERENCES:
+        // (I)BLAST legacy routine HBSRF
+
+        using namespace DataGlobals;
+        using namespace DataEnvironment;
+        using ConvectionCoefficients::InitExteriorConvectionCoeff;
+        using ConvectionCoefficients::InitInteriorConvectionCoeffs;
+        using ConvectionCoefficients::SetExtConvectionCoeff;
+        using DataLoopNode::Node;
+        using DataZoneEquipment::ZoneEquipConfig;
+        using General::RoundSigDigits;
+        using namespace Psychrometrics;
+        using DataSizing::CurOverallSimDay;
+
+        Real64 const IterDampConst(5.0);       // Damping constant for inside surface temperature iterations
+        int const ItersReevalConvCoeff(30);    // Number of iterations between inside convection coefficient reevaluations
+        Real64 const MaxAllowedDelTemp(0.002); // Convergence criteria for inside surface temperatures
+        int const MaxIterations(500);          // Maximum number of iterations allowed for inside surface temps
+        Real64 const SmallNumber(0.0001);      // avoid numerical junk causing problems?
+        // in the CondFD relaxation factor.
+        static std::string const rhoAirZone("RhoAirZone");
+        static std::string const wsurf("Wsurf");
+        static std::string const HBSurfManInsideSurf("HB,SurfMan:InsideSurf");
+        static std::string const Inside("Inside");
+        static std::string const BlankString;
+
+        Real64 AbsInt;     // Solar absorptance of inside movable insulation
+        bool Converged;    // .TRUE. if inside heat balance has converged
+        Real64 F1;         // Intermediate calculation value
+        Real64 HMovInsul;  // "Convection" coefficient of movable insulation
+        Real64 MaxDelTemp; // Maximum change in surface temperature for any
+        //  opaque surface from one iteration to the next
+        static Array1D<Real64> TempInsOld; // Holds previous iteration's value for convergence check
+
+        int OtherSideSurfNum;     // Surface number index for other side of an interzone partition
+        static int MinIterations; // Minimum number of iterations for the inside heat balance
+        static int ErrCount(0);
+        Real64 Ueff; // 1 / effective R value between TDD:DOME and TDD:DIFFUSER
+
+        int ZoneEquipConfigNum;
+        int NodeNum;
+        Real64 SumSysMCp;  // Zone sum of air system MassFlowRate*Cp
+        Real64 SumSysMCpT; // Zone sum of air system MassFlowRate*Cp*T
+        Real64 MassFlowRate;
+        Real64 NodeTemp;
+        Real64 CpAir;
+        static Array1D<Real64> RefAirTemp; // reference air temperatures
+        static bool MyEnvrnFlag(true);
+        static int InsideSurfErrCount(0);
+        static int WarmupSurfTemp;
+        static int TimeStepInDay(0); // time step number
+
+        if (calcHeatBalanceInsideSurfCTFOnlyFirstTime) {
```

Comment:
This block of one-time stuff should ultimately move somewhere else, but kept the same pattern here as in `CalcHeatBalanceInsideSurf2` for now. This keeps all the related changes close together.